### PR TITLE
Backport all CI matrix changes from 7.0 to reelase/6.0

### PR DESCRIPTION
**Pull request by aik-jahoda**
Most of the changes are tracked there: https://github.com/dotnet/runtime/issues/57947#issuecomment-917441233 

There is no product change so impact of the change is only in CI not in product itself.

### CODE REVIEW

**Comment by aik-jahoda** [comment]
Passing tests: https://dev.azure.com/dnceng/public/_build/results?buildId=1503206&view=results

**Review by danmoseley — APPROVED** [review]
I can't review detail of YAML changes, but I approve the change.

**Comment by danmoseley** [comment]
No tactics approval needed.

File: eng/pipelines/runtime-community.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+trigger: none
```
Comment (safern) [inline eng/pipelines/runtime-community.yml:1]: This template doesn't make sense to port it as it doesn't even have triggers for release/* branches.

**Review by safern — APPROVED** [review]
Other than one comment, looks good.

**Comment by aik-jahoda** [comment]
@safern 

> Other than one comment, looks good.

I will revert whole 4199aa1 from the backport.

**Comment by aik-jahoda** [comment]
Passing tests: https://dev.azure.com/dnceng/public/_build/results?buildId=1510275&view=results

There is one unrelated failure.

**Pull request merged**